### PR TITLE
chore(github-action): Refine GitHub Actions criteria for builds, pull requests and release

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder niladri.halder@mayadata.io

The build.yml and pull_request.yml workflows will ignore pushes made to `example/` and `changelogs/` directories, all .md files, LICENSE and MAINTAINERS files.

Added criteria to ignore build workflows on release.

### CODE REVIEW

**Comment by niladrih** [comment]
Related PRs:
https://github.com/openebs/node-disk-manager/pull/562
https://github.com/openebs/cstor-operators/pull/271
https://github.com/openebs/dynamic-localpv-provisioner/pull/41
https://github.com/openebs/zfs-localpv/pull/306
https://github.com/openebs/lvm-localpv/pull/42
https://github.com/openebs/jiva-operator/pull/66
https://github.com/openebs/maya/pull/1797
https://github.com/openebs/cstor-csi/pull/151
https://github.com/openebs/jiva/pull/350
https://github.com/openebs/libcstor/pull/85
https://github.com/openebs/m-exporter/pull/14

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,10 +13,23 @@
...
+on:
+  create:
+  push:
+    branches:
+      - 'master'
+      - 'v*'
+    paths-ignore:
+      - '*.md'
+      - 'changelogs/**'
+      - 'example/**'
```
Comment (akhilerm) [inline .github/workflows/build.yml:25]: Is the example directory used for any testing purpose or is it purely for examples?

**Review by akhilerm — APPROVED** [review]
A comment on the usage of example directory.

File: .github/workflows/build.yml (hunk 1 above)
Comment (niladrih) [reply .github/workflows/build.yml:25]: The files for testing are in script/ directory. example/ hasn't been used in any of the GitHub Action workflows.

**Review by shubham14bajpai — APPROVED** [review]
@niladrih Can you update the Build & Push step in github workflows to use `uses: docker/build-push-action@v2` instead of Makefile.buildx targets. Also add the Quay login. Ref openebs/cstor-operator.

**Review by shubham14bajpai — APPROVED** [review]
LGTM

**Review by mynktl — APPROVED** [review]

**Review by akhilerm — APPROVED** [review]
LGTM

**Comment by akhilerm** [comment]
@kmova Need to add the QUAY tokens to this repository.

**Pull request merged**